src/main/java/qna/service/QnAService.java

```diff
@@ -35,24 +32,24 @@ public Question findQuestionById(Long id) {
     @Transactional
     public void deleteQuestion(User loginUser, long questionId) throws CannotDeleteException {
         Question question = findQuestionById(questionId);
+        validateUser(question, loginUser);
+        question.delete();
```

Comment:
```suggestion
        question.delete(loginUser);
```
이 두가지 메서드를 비교해보시면 좋을 것 같아요. 
- 만약 QnAService가 아닌 다른 서비스에서 question.delete()를 호출했는데 로그인 한 유저가 작성자와 다르면 어떤 일이 벌어질까요?
- 서비스를 테스트 하기 위해서는 mocking이나 실제 db 주입 등 다른 기술들이 추가로 필요합니다. 반면에 도메인만 테스트하는 경우에는 이런 기술들 없이 테스트가 나타내고자하는 내용을 깔끔하게 나타낼 수 있을 것 같아요.

Reply:
유저를 벨리데이션 하는 로직은 말씀대로, 지금 구조의 delete() 메서드를 호출하기전에 여러 서비스에서 중복적으로 흩어질수 있는 로직인것 같아요.
벨리데이션 하는 로직은 question 클래스 내부로 옮기고, delete(loginUser) 메서드 내부에서 자체적으로 파라미터인 loginUser에 대한 벨리데이션을 수행하도록 다시 바꾸어봤습니다.